user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md

```diff
@@ -0,0 +1,138 @@
+---
+uid: GQI_Extensions_Logging
+---
+
+# Logging
+
+From DataMiner 10.4.5/10.5.0 onwards<!-- RN 39043 -->, [ad hoc data sources](xref:Configuring_an_ad_hoc_data_source_in_a_query) and [custom operators](xref:GQI_Custom_Operator) can log their own messages and exceptions within GQI, complementing the existing [GQI core logging functionality](xref:GQI_Logging) (available from DataMiner 10.4.0/10.4.4).
+
+## Prerequisites
+
+To enable logging, the GQI extension (either an ad hoc data source or a custom operator) must implement the [IGQIOnInit](xref:GQI_IGQIOnInit) interface.
+
+This building block adds the [OnInit](xref:GQI_IGQIOnInit#oninitoutputargs-oninitoninitinputargs-args) lifecycle method, granting access to the [IGQILogger](xref:GQI_IGQILogger) object through the [Logger property](xref:GQI_OnInitInputArgs#properties) of the [OnInitInputArgs](xref:GQI_OnInitInputArgs) class.
+
+Throughout its lifespan, the extension uses the [IGQILogger](xref:GQI_IGQILogger) object to log messages and exceptions.
```

Comment:
The goal here was to stress how *you, the extension creator*,  can take action to log messages. Phrasing it like this gives the impression that there is more info needed to log your own messages but that's not the case.

```suggestion
Throughout its lifespan, the extension can use the [IGQILogger](xref:GQI_IGQILogger) methods to log messages and exceptions.
```